frontend/src/components/CartItem.tsx

```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import { Button, ListItem, ListItemText } from "@mui/material";
+
+interface CartItemProps {
+  id: string;
+  name: string;
+  price: number;
+  quantity: number;
+  removeFromCart: (id: string) => void;
+}
+
+const CartItem: React.FC<CartItemProps> = ({
+  id,
+  name,
+  price,
+  quantity,
```

Comment:
you can try subsititue these fields with one interface here, this will make code cleaner